server/groups/class.lua

```diff
@@ -85,6 +88,15 @@ function CGroup:set(player, grade)
     if currentGrade then
         if currentGrade == grade then return end
         self:remove(player, currentGrade)
+    else
+        if self.type and self.unique_type == 1 then
+            for _ in pairs(player.private.groups) do
+                local groupCompare = Ox.GetGroup(_)
+                if groupCompare.type == self.type then
+                    error(("Attempted to add multiple unique group '%s' with '%s' for player.%s"):format(self.type, self.name, player.source))
```

Comment:
Throwing an error can be a bit awkward when an error didn't actually occur, especially when pcall isn't a common practice in FiveM.

A standardised warning system has finally been added to FiveM so you can use `warn(message)` instead without ending the current thread, then we need to return a boolean for success/failure.